src/autoscaler/eventgenerator/generator/evaluator.go

```diff
@@ -70,27 +70,27 @@ func (e *Evaluator) doEvaluate(triggerArray []*model.Trigger) {
 			return
 		}
 		for _, appMetric := range appMetricList {
-			if appMetric.Value < 0 {
+			if appMetric.Value == nil {
 				e.logger.Info("should not send trigger alarm to scaling engine because there is nil-value metric", lager.Data{"trigger": trigger, "appMetric": appMetric})
```

Comment:
When it get an empty metric list from metric-collector, it sets the avg AppMetric.Value to nil. So here check the nil value. If it is nil , no trigger should be sent.